texts/0005-api-refactor.md

```diff
@@ -0,0 +1,79 @@
+- Feature Name: v2_api_refactor
+- Start Date: 2023-03-24
+- RFC PR: [tauri-apps/rfcs#0005](https://github.com/tauri-apps/rfcs/pull/10)
+
+# Summary
+
+With the plugin interface being capable of integrating with mobile-specific functionalities (see [RFC #0005](https://github.com/tauri-apps/rfcs/pull/9)), it makes sense for each existing API module to be rewritten as a plugin in order to use this new interface to implement current desktop API for Android and iOS. For instance, APIs like dialog, which uses [`rfd`](https://github.com/PolyMeilex/rfd), a desktop-only crate, can use the new mobile plugin structure to implement file and message dialogs for mobile.
+
+# Motivation
+
+The main motivation is actually implementing the existing APIs for mobile targets. This gets complicated with the current codebase, which already includes a lot of conditional compilation checks for both allowlist and platform detection. Separating each API module as its own plugin enhances maintainability as each package has a limited scope.
+
+Additionally, this change simplifies future audits since the scope of changes are much more restricted and we can easily check which functionalities were actually changed, instead of inspecting changes in a monolith core crate that is always evolving.
+
+# Guide-level explanation
+
+Two types of plugins will be defined: core plugins and regular (optional? external?) plugins. Core plugins are actually used by the `tauri` crate so they are defined in the crate itself and automatically configured and enabled, users will see these plugins as regular Rust modules that just export types and functions to be directly used. That is the case for the `path` and `file_system` plugins. The other APIs (`cli`,  `dialog`, `http`, `notification` and `process`) are optional and will be defined in the [plugins-workspace repo](https://github.com/tauri-apps/plugins-workspace) along with new plugins introduced in v2 like `camera` and `geolocation`.
```

Comment:
I like that you've put in this distinction between core and the other plugins. I was also trying to figure out what to call the "other" ones and haven't come up with anything quite yet.